packages/clients/src/internal/interceptors/types.ts

```diff
@@ -0,0 +1,17 @@
+export interface RequestInterceptor {
+  (request: Readonly<Request>): Request | Promise<Request>
+}
+
+export interface ResponseInterceptor {
+  (response: Readonly<Response>): Response | Promise<Response>
+}
+
+export interface ResponseErrorInterceptor {
+  (request: Request, error: unknown): unknown | Promise<unknown>
```

Comment:
can we said that the return should be (void | Promise<void>) | unknown | Promise<unknown>  ?

And maybe add a comment to explain why ( allow typescript to have some input :) )

Reply:
Didn't find a "good way" to achieve "unknown minus vois". I think properly commenting the usage should be enough as it's an uncommon usage of the SDK.